### PR TITLE
RedundantParens: add infixSide parameter to control readability

### PR DESCRIPTION
**Pull request by kitbellew**

### CODE REVIEW

File: scalafmt-tests/src/test/resources/rewrite/AvoidInfix.stat [hunk 1]
```diff
@@ -388,7 +388,7 @@ object a {
 }
 >>>
 object a {
-  (a(b("c")).over(d) / (e + 1)).as(f)
+  ((a(b("c")).over(d)) / (e + 1)).as(f)
```
Comment (tgodzik) [inline scalafmt-tests/src/test/resources/rewrite/AvoidInfix.stat:391]: Should the default behaviour change?

File: scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat [hunk 2]
```diff
@@ -267,8 +267,8 @@ object a {
 }
 >>>
 object a {
-  private val foo = (bar == baz
-    || bar == baz
+  private val foo = ((bar == baz
```
Comment (tgodzik) [inline scalafmt-tests/src/test/resources/rewrite/RedundantParens.stat:270]: Same here, we should probably avoid chanigng the default behavior?

File: docs/configuration.md [hunk 3]
```diff
@@ -2801,6 +2801,25 @@ for {
...
+#### `RedundantParens`: `infixSide`
+
+> Since v3.5.4.
+
+Parameter `rewrite.redundantParens.infixSide` controls how the rule applies
+to expressions which are part of an outer infix expression (either left- or
+right-hand side). Can take the following values:
+
+- `null` (default): rewrites only simple expressions (literals or identifiers)
+- `some`: additionally, rewrites
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply docs/configuration.md:2813]: will do

File: scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala [hunk 4]
```diff
@@ -98,6 +98,23 @@ class RedundantParens(ftoks: FormatTokens) extends FormatTokensRewrite.Rule {
           case p: Term.If =>
             style.dialect.allowSignificantIndentation && p.cond == t &&
             ftoks.tokenBefore(p.thenp).left.is[Token.KwThen]
+          case InfixApp(pia) if !infixNeedsParens(pia, t) =>
+            t match {
+              case InfixApp(tia) =>
+                style.rewrite.redundantParens.infixSide.exists {
+                  case RedundantParensSettings.InfixSide.many =>
+                    tia.op.value == pia.op.value ||
+                    tia.precedence <= 4 ||
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantParens.scala:107]: of course

File: scalafmt-tests/src/test/resources/rewrite/AvoidInfix.stat (hunk 1 above)
Comment (kitbellew) [reply scalafmt-tests/src/test/resources/rewrite/AvoidInfix.stat:391]: actually, it's simply reverting back to pre-3.5.3 behavior; same with the other case.

File: docs/configuration.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kitbellew) [reply docs/configuration.md:2827]: done

**Review by tgodzik — APPROVED** [review]
LGTM!

**Pull request merged**